force-app/main/default/classes/UTIL_DuplicateMgmt.cls

```diff
@@ -0,0 +1,157 @@
+/*
+    Copyright (c) 2022, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2022
+* @group Utilities
+* @description Utility methods to faciliate using the built-in findDuplicates function.
+*/
+public with sharing class UTIL_DuplicateMgmt {
+
+    /**
+    * @description Returns map of duplicate matching details for the records passed in.  Note:
+    * Will throw error if more than 50 SObjects are passed in.
+    * @param List<sObject> list of Sobjects that duplicates will be located for
+    * @return Map<SObject, BDI_DuplicateService.MatchDetail[]>
+    */
+    public static Map<SObject, MatchDetail[]> findDuplicates(List<SObject> records) {
```

Comment:
@voduyemi It doesn't necessarily need to be static although in general I make utility class methods static unless they need to be non-static.  I had to make the findDuplicatesBulk class non-static so that I could mock it correctly.  

For the 50 max limit, the findDuplicates core method already handles that error cleanly so there is no reason to add another layer of error messages/handling.  I actually removed one I had in there since it was redundant.

Reply:
We might as well make it private since at most it would save a little processing and could be dangerous.